### ISSUE TITLE
"singularity" is a reserved word in a Snakefile

### ISSUE THREAD

**Issue by fburic** [snakemake/snakemake#614]
**Snakemake version**
5.24

**Describe the bug**
If one defines a variable in the Snakefile as
```python
singularity = config['singularity_string']  # string that holds which image to run and some flags

# its use would be like 
# shell:
#   "{singularity} COMMAND"
```

One gets the error: `Colon expected after keyword global_container.`
And this is a parsing error - the snakefile does not actually run.

Changing the name of the variable to e.g. `sing` works.

This is not a big issue but I'm worried that the scope of such variables somehow gets mixed with Snakemake's internals.

**Comment by sschmeier** [comment]
I think the problem stes from the fact that snakemake had a argument `singularity` with which you could specifyy a global singularity container for the wholee workflow. e.g.

```python
singularity: "docker//bla..."

rule one:
...
````

have you tried calling the var something else? Should work.

**Comment by fburic** [comment]
Yes, calling the variable something else solves the immediate problem, as I mentioned.

If it's an old keyword, I suppose there should be an effort to clean those up and perhaps have more informative errors should an user try to use a reserved word.

**Comment by Hocnonsense** [comment]
Yes, `singularity` is just an legacy alias of `container`, and as of Snakemake ≥9, it's only kept for backwards compatibility with older workflows. It's likely to be deprecated or removed in a future major release.

> This is not a big issue but I'm worried that the scope of such variables somehow gets mixed with Snakemake's internals.
That's a valid observation, but no need to worry too much. Most reserved keywords in Snakemake are handled at the DSL parsing stage, so if there's a conflict, the Snakefile will fail to parse long before execution. This makes such issues obvious and easy to catch.
